src/Agent/NewRelic/Agent/Core/Attributes/AttributeValueCollection.cs

```diff
@@ -138,6 +148,26 @@ public IEnumerable<IAttributeValue> GetAttributeValues(AttributeClassification c
             return result;
         }
 
+        public IDictionary<string, object> GetAllAttributeValuesDic()
+        {
+            var result = new Dictionary<string, object>();
+            foreach (var classification in _allClassifications)
+            {
+                foreach (var attribVal in GetAttribValuesImpl(classification))
+                {
+                    if (classification == AttributeClassification.UserAttributes && _reservedProperties.Contains(attribVal.AttributeDefinition.Name) )
```

Comment:
Rather than having explicit reserved properties, maybe it would be cleaner to loop through the classifications in increasing priority order? user -> intrinsic -> agent?

Reply:
I think that this recommendation is simpler but probably deserves a comment and some unit tests otherwise someone might change the order in the future.

Reply:
I updated the existing "_allClassifications" to use the order you mentioned.  I double checked and it everywhere else it was used did not depend on the order.  I also added a few comments with the order and a new test that confirms the order works.